src/SimpleReplay/extract/extractor/extractor.py

```diff
@@ -111,11 +149,17 @@ def save_logs(self, logs, last_connections, output_directory, connections, start
             )
             pathlib.Path(output_directory).mkdir(parents=True, exist_ok=True)
 
-        sql_json, missing_audit_log_connections, replacements = self.get_sql_connections_replacements(last_connections,
-                                                                                                      log_items)
-
-        with gzip.open(archive_filename, "wb") as f:
-            f.write(json.dumps(sql_json, indent=2).encode("utf-8"))
+        sql_json, missing_audit_log_connections, replacements,stored_procedures_map = self.get_sql_connections_replacements(last_connections,
+                                                                                                      log_items) 
+        sql_json_with_stored_procedure = asyncio.run(Extractor.get_stored_procedures(self,start_time=self.config['start_time'],end_time=self.config['end_time'],username=self.config['master_username'],stored_procedures_map=stored_procedures_map,sql_json=sql_json))
+        logger.info(f'The total length of stored procedures found are : {len(stored_procedures_map)}')
+        import pdb;pdb.set_trace()
+        if self.config['replay_stored_procedures']:
```

Comment:
I would skip even doing the query if the config variable is false. Also use `self.config.get('replay_stored_procedures')`

Reply:
okay done